src/SFA.DAS.LoginService.Web/Infrastructure/IdentityServerStartupExtensions.cs

```diff
@@ -63,4 +75,19 @@ public static void AddIdentityServer(this IServiceCollection services, ILoginCon
             }
         }
     }
+
+    public interface IManagedIdentityTokenProvider
+    {
+        string GetAccessToken();
+    }
+
+    public class ManagedIdentityAzureTokenProvider : IManagedIdentityTokenProvider
+    {
+        public string GetAccessToken()
+        {
+            return new AzureServiceTokenProvider()
+                .GetAccessTokenAsync("https://database.windows.net/")
+                .GetAwaiter().GetResult();
+        }
+    }
```

Comment:
It's not something I worry about but I believe SFA coding standard is to put classes in separate files. There are several times when I don't think it's appropriate, but it's up to you